### ISSUE TITLE
SignalR js client doesn't fallback to LongPolling when it fails to connect with server side events

### ISSUE THREAD

**Issue by Hrydberg** [dotnet/aspnetcore#20500]
### Describe the bug
SignalR js client doesn't fallback to LongPolling when it fails to connect with server side events.

### To Reproduce
Setup the scenario described in #20498 

The following is logged and it doesn't try to connect with LongPolling after SSE fails.
![image](https://user-images.githubusercontent.com/3148297/78364353-99556280-75bd-11ea-8d6d-13b68f1f746d.png)

These are the network calls made.
![image](https://user-images.githubusercontent.com/3148297/78364458-c30e8980-75bd-11ea-8ccb-bab97cd70e22.png)

### Second scenario
I also had the issue in another scenario where I think a local firewall is messing with the connection where it won't fallback to LongPolling.
There this is logged.
![image](https://user-images.githubusercontent.com/3148297/78364756-49c36680-75be-11ea-86aa-44b5c6f7ae9a.png)

And these network calls are made.
![image](https://user-images.githubusercontent.com/3148297/78365258-1fbe7400-75bf-11ea-80c7-ad50f7c5b661.png)

When I use an old version of ASP.Net SignalR on the same server the fallback to LongPolling works as expected.
![image](https://user-images.githubusercontent.com/3148297/78364986-b0e11b00-75be-11ea-9c74-23c184ebfde3.png)

### Further technical details
- ASP.NET Core version 3.1

**Comment by davidfowl** [comment]
Are you using https? In both cases the connection was successfully made but then it failed to communicate afterwards (likely something in between buffering responses).

It may be better in these scenarios to disable SSE completely.

**Comment by Hrydberg** [comment]
No I am not using https.
But shouldn't it fallback to LongPolling automatically like the old SignalR client did?

**Comment by davidfowl** [comment]
We’ll investigate why it’s happening for sure, old signalr makes some tradeoffs we won’t be willing to make in the new version though (which greatly affect things like connection establishment time). 

Are you in a situation where you cannot use websockets? Or weee you just testing other transports to see the behavior.

**Comment by Hrydberg** [comment]
We have some clients that are running our application on old Windows Server 2008 R2 where websockets isn't available.
Thank you for investigating this.

**Comment by BrennanConroy** [comment]
SSE has already connected successfully which will disable transport fallback. The real issue is https://github.com/dotnet/aspnetcore/issues/20498 which we will definitely take a look at.

Also, you should consider disabling SSE on the client. In fact, we're even considering removing SSE in a future release.

**Comment by shirhatti** [comment]
Triage: In the old SignalR client, we waited for a message to be received before disabling transport fallback. However, this is low priority

**Comment by davidfowl** [comment]
This feels like the proxy in the middle buffering problem because there's no https.

**Comment by BrennanConroy** [comment]
Yes it's a proxy issue, but we could try to change our behavior here to fallback anyways. But because we assume "start" or "connect" succeeding is a successful connection instead of waiting for any bytes, we stop the fallback logic.

**Comment by davidfowl** [comment]
Client work is work * 3 right now. This one might be small, but you tell me.

**Comment by BrennanConroy** [comment]
It probably isn't small. It's more internal state management which is always tricky.

**Comment by rajeevBlr** [comment]
We are also facing this issue. After SSE fails, Signalr does not fallback to LongPolling. @BrennanConroy if possible please increase the severity of the bug.

**Comment by MatthiasLichter** [comment]
Do you have any updates on this issue? It is frustrating to search for and explain.

**Comment by lawrencetaur** [comment]
We had the same issue trying to connect React with 3 pods of our app connected to using Redis backplane served by Ingress Nginx on Kubernetes. 

Here are the nginx ingress annotations we used:

```
    nginx.ingress.kubernetes.io/affinity: cookie
    nginx.ingress.kubernetes.io/ssl-redirect: "false"
    nginx.ingress.kubernetes.io/session-cookie-hash: sha1
    nginx.ingress.kubernetes.io/session-cookie-name: REALTIMESERVERID
    nginx.ingress.kubernetes.io/affinity-mode: persistent
    nginx.ingress.kubernetes.io/session-cookie-secure: "true"
    nginx.ingress.kubernetes.io/session-cookie-samesite: "None"
```
    
`session-cookie-secure` and `session-cookie-samesite` did the trick because our app was loading inside an iframe via Shopify.
